### PR TITLE
Remove GCP Broker provider

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- Remove GCP broker provider tool

**Related issue(s)**
Related: https://github.com/kyma-project/kyma/issues/4782

### CODE REVIEW

**Comment by adamwalach** [comment]
The gcp-broker-provider image is also used in azure and aws brokers: https://github.com/kyma-project/bundles/blob/master/bundles/azure-service-broker-0.0.1/chart/azure-service-broker/values.yaml#L82
I think we can safely replace it with `bitnami/kubectl:1.12-debian-9`

File: docs/migration-guides/1.3-1.4.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Migrate from v1.3 to v1.4
+
+>**TIP:** To learn more about upgrading Kyma, read [this](https://kyma-project.io/docs/master/root/kyma/#installation-upgrade-kyma) document.
+
+## Service Management
+### Google Cloud Platform Service Broker addon
+
+As the Google Cloud Platform Service Broker is being [deprecated](https://cloud.google.com/kubernetes-engine/docs/concepts/google-cloud-platform-service-broker), we changed this addon to use different implementation. If you have already installed the Google Cloud Platform Service Broker in Kyma 1.3, you see this addon with two versions.
```
Comment (klaudiagrz) [inline docs/migration-guides/1.3-1.4.md]:
```suggestion
As the Google Cloud Platform (GCP) Service Broker is being deprecated, we have changed this addon to use a different implementation. This is why you see this addon in two versions if you have already installed the GCP Service Broker in Kyma 1.3. 
```

File: docs/migration-guides/1.3-1.4.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+If you haven't installed any GCP services in a given Namespace, delete the Google Cloud Platform Service Broker instance in this Namespace.
```
Comment (klaudiagrz) [inline docs/migration-guides/1.3-1.4.md]:
```suggestion
If you still haven't installed any GCP services in your Namespace, delete the GCP Service Broker instance in this Namespace.
```

Comment (klaudiagrz) [reply docs/migration-guides/1.3-1.4.md]: Do we need this sentence? We state it again below in "Consider cleaning up...".

File: docs/migration-guides/1.3-1.4.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+You can still manage the existing GCP services using Kyma until November 5, 2019. On or after that time, you will be able to manage services using the API, the command line or Google Cloud Platform Console.
+Consider cleaning up not needed GCP Service Bindings, Service Instances and removal Google Cloud Platform Service Broker instances. Follow these steps:
```
Comment (klaudiagrz) [inline docs/migration-guides/1.3-1.4.md]:
```suggestion
Consider removing GCP Service Bindings, Service Instances, and GCP Service Broker instances that you don't need. To do so, follow these steps:
```

File: docs/migration-guides/1.3-1.4.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+1. Delete all bindings to GCP Service Instances.
+2. Delete all GCP Service Instances.
+3. Delete all Google Cloud Platform Service Broker Service Instances.
```
Comment (klaudiagrz) [inline docs/migration-guides/1.3-1.4.md]:
```suggestion
3. Delete all GCP Service Broker instances.
```

File: docs/migration-guides/1.3-1.4.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+If you need a new instance of GCP services, provision Google Cloud Platform Service Broker 0.0.2 and use its service classes.
```
Comment (klaudiagrz) [inline docs/migration-guides/1.3-1.4.md]:
```suggestion
If you need a new instance of GCP services, provision GCP Service Broker 0.0.2 and use its Service Classes.
```

Comment (klaudiagrz) [reply docs/migration-guides/1.3-1.4.md]: ?

File: docs/migration-guides/1.3-1.4.md (hunk 2 above)
Comment (piotrmiskiewicz) [reply docs/migration-guides/1.3-1.4.md]: I think we need this sentence.

File: docs/migration-guides/1.3-1.4.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+>**TIP:** To learn more about upgrading Kyma, read [this](https://kyma-project.io/docs/master/root/kyma/#installation-upgrade-kyma) document.
+
+## Service Management
+
+As the Google Cloud Platform (GCP) Service Broker is being deprecated, we have removed this addon. If you have already installed the GCP Service Broker in Kyma 1.3 you see this addon. 
+If you still haven't installed any GCP services in your Namespace, delete the GCP Service Broker instance in this Namespace.
+
+You can still manage the existing GCP services using Kyma until November 5, 2019. On or after that time, you will be able to manage services using the API, the command line or Google Cloud Platform Console.
+Consider removing GCP Service Bindings, Service Instances, and GCP Service Broker instances that you don't need. To do so, follow these steps:
```
Comment (klaudiagrz) [inline docs/migration-guides/1.3-1.4.md]:
```suggestion
If you have installed the GCP Service Broker addon but haven't provisioned any of its services in your Namespace, delete the GCP Service Broker instance from the Namespace. If you have provisioned any GCP Service Broker services, consider removing the ones that you don't need. To do so, follow these steps:
```

File: docs/migration-guides/1.3-1.4.md [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+1. Delete all bindings to GCP Service Instances.
+2. Delete all GCP Service Instances.
+3. Delete all GCP Service Broker instances.
+
+>**NOTE:** The charges incurred by the remaining resources created through the GCP Service Broker will continue to accrue. For more information, read the official Google Cloud [documentation](https://cloud.google.com/kubernetes-engine/docs/concepts/google-cloud-platform-service-broker).
```
Comment (klaudiagrz) [inline docs/migration-guides/1.3-1.4.md]:
```suggestion
The charges incurred by the remaining resources created through the GCP Service Broker will continue to accrue. For more information, read the official Google Cloud [documentation](https://cloud.google.com/kubernetes-engine/docs/concepts/google-cloud-platform-service-broker).
```

**Pull request merged**